pom.xml

```diff
@@ -68,23 +68,45 @@
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
           <version>3.8.1</version>
+          <configuration>
+            <release>${javaVersion}</release>
+            <jdkToolchain>
+              <!-- Require at most JDK 11 because it is the last version 
+                supporting compiling Java 1.6, see https://bugs.openjdk.java.net/browse/JDK-8028563 -->
+              <!-- When Gson increases lowest supported Java version, can 
+                change this to an open version range (e.g. `[11,)`) -->
+              <version>[9,11]</version>
```

Comment:
Not completely sure why this works for the GitHub CI workflow (or rather why it is ignored?). For Surefire plugin it caused build failures, see https://github.com/Marcono1234/gson/runs/4146407481.